utils/db_utils.py

```diff
@@ -131,123 +131,153 @@ def query_uuids(start_date: str, end_date: str, tz: str):
+                    or not isinstance(md["distance"], dict)
+                    # If 'md' is a dictionary and 'distance' is a valid key pointing to a dictionary:
+                    else (
+                        # Get the maximum value from 'md["distance"]' using the values of 'md["distance"].get' as the key for 'max'.
+                        # This operation only happens if the length of 'md["distance"]' is greater than 0.
+                        # Otherwise, return "INVALID".
+                        max(md["distance"], key=md["distance"].get)
+                        if len(md["distance"]) > 0
+                        else "INVALID"
+                    )
+                )
+    
+                df["data.primary_sensed_mode"] = df.cleaned_section_summary.apply(get_max_mode_from_summary)
+                df["data.primary_predicted_mode"] = df.inferred_section_summary.apply(get_max_mode_from_summary)
+                if 'ble_sensed_summary' in df.columns:
+                    df["data.primary_ble_sensed_mode"] = df.ble_sensed_summary.apply(get_max_mode_from_summary)
+                else:
+                    logging.debug("No BLE support found, not fleet version, ignoring...")
             esdsq.store_dashboard_time(
-                "admin/db_utils/query_confirmed_trips/drop_metadata_columns",
+                "admin/db_utils/query_confirmed_trips/add_primary_modes",
                 stage5_timer
             )
-            
-            # Stage 6: Drop or Modify Excluded Columns
+    
+            # Stage 6: Expand User Inputs
             with ect.Timer() as stage6_timer:
-                # Drop or modify excluded columns
-                for col in constants.EXCLUDED_TRAJECTORIES_COLS:
-                    if col in df.columns:
-                        df.drop(columns=[col], inplace=True)
-            esdsq.store_dashboard_time(
-                "admin/db_utils/query_confirmed_trips/drop_or_modify_excluded_columns",
-                stage6_timer
-            )
-            
-            # I dont think we even implemented this..
-            # need fix asap
-            # Stage 7: Handle 'background/location' Key
-            with ect.Timer() as stage7_timer:
-                # Check if 'background/location' is in the key_list
-                if 'background/location' in key_list:
-                    if 'data.mode' in df.columns:
-                        # Set the values in data.mode to blank ('')
-                        df['data.mode'] = ''
-                else:
-                    # Map mode to its corresponding string value
-                    df['data.mode_str'] = df['data.mode'].apply(
-                        lambda x: ecwm.MotionTypes(x).name if x in set(enum.value for enum in ecwm.MotionTypes) else 'UNKNOWN'
-                    )
-            esdsq.store_dashboard_time(
-                "admin/db_utils/query_confirmed_trips/handle_background_location_key",
-                stage7_timer
-            )
```

Comment:
This was the main culprit

Reply:
What in this in the culprit?

Reply:
I thought we had the key list for trips tab just like the one in trajectories but looking back and previous commits/versions showed that we did not

Reply:
To be honest I think I just mixed up query_trajectories and query_trips; a late night weekend blunder if you will